src/util/misc/boundingBoxFromPoints.ts

```diff
@@ -8,31 +7,23 @@ import type { TBBox } from '../../typedefs';
  * @return {Object} Object with left, top, width, height properties
  */
 export const makeBoundingBoxFromPoints = (points: XY[]): TBBox => {
-  if (points.length === 0) {
-    return {
-      left: 0,
-      top: 0,
-      width: 0,
-      height: 0,
-    };
-  }
-
-  const { min, max } = points.reduce(
-    ({ min, max }, curr) => {
-      return {
-        min: min.min(curr),
-        max: max.max(curr),
-      };
-    },
-    { min: new Point(points[0]), max: new Point(points[0]) },
-  );
+  let left = 0,
+    top = 0,
+    width = 0,
+    height = 0;
 
-  const size = max.subtract(min);
+  for (let i = 0, len = points.length; i < len; i++) {
+    const p = points[i];
+    if (p.x > width || i === 0) width = p.x;
+    if (p.x < left || i === 0) left = p.x;
+    if (p.y > height || i === 0) height = p.y;
+    if (p.y < top || i === 0) top = p.y;
+  }
```

Comment:
the code we had before was slower because in general reduces has to:
- create a function stack for each item in the loop
- it was creating 2 new point instances every item in the loop
- it was creating a new object to return every item in the loop
- it triggerd gc for long loops

The newer code is dumber, even dumber of what we had before in fabric 5, but is just reassigning over and over a bunch of numbers and not creating any new isntances to garbage collect later

Reply:
Nice job 👍